examples/preprocessing/plot_all_scaling.py

```diff
@@ -67,15 +67,25 @@
 
 print(__doc__)
 
-dataset = fetch_california_housing()
+dataset = fetch_california_housing(as_frame=True)
 X_full, y_full = dataset.data, dataset.target
 
-# Take only 2 features to make visualization easier
-# Feature of 0 has a long tail distribution.
-# Feature 5 has a few but very large outliers.
-
-X = X_full[:, [0, 5]]
+feature_mapping = {
+    'MedInc': 'Median income in block',
+    'HousAge': 'Median house age in block',
+    'AveRooms': 'Average number of rooms',
+    'AveBedrms': 'Average number of bedrooms',
+    'Population': 'Block population',
+    'AveOccup': 'Average house occupancy',
+    'Latitude': 'House block latitude',
+    'Longitude': 'House block longitude'
+}
 
+# Take only 2 features to make visualization easier
+# Feature MedInc has a long tail distribution.
+# Feature AveOccup has a few but very large outliers.
+features = ['MedInc', 'AveOccup']
+X = X_full[features].values
```

Comment:
We can avoid this transform be changing the indexing in the plotting function.
The diff will be something like this:

```diff
diff --git a/examples/preprocessing/plot_all_scaling.py b/examples/preprocessing/plot_all_scaling.py
index 6aa205dd14..13ff353692 100755
--- a/examples/preprocessing/plot_all_scaling.py
+++ b/examples/preprocessing/plot_all_scaling.py
@@ -85,7 +85,7 @@ feature_mapping = {
 # Feature MedInc has a long tail distribution.
 # Feature AveOccup has a few but very large outliers.
 features = ['MedInc', 'AveOccup']
-X = X_full[features].values
+X = X_full[features]
 distributions = [
     ('Unscaled data', X),
     ('Data after standard scaling',
@@ -168,7 +168,9 @@ def plot_distribution(axes, X, y, hist_nbins=50, title="",
 
     # The scatter plot
     colors = cmap(y)
-    ax.scatter(X[:, 0], X[:, 1], alpha=0.5, marker='o', s=5, lw=0, c=colors)
+    ax.scatter(
+        X[features[0]], X[features[1]], alpha=0.5, marker='o', s=5, lw=0, c=colors
+    )
 
     # Removing the top and the right spine for aesthetics
     # make nice axis layout
@@ -181,13 +183,13 @@ def plot_distribution(axes, X, y, hist_nbins=50, title="",
 
     # Histogram for axis X1 (feature 5)
     hist_X1.set_ylim(ax.get_ylim())
-    hist_X1.hist(X[:, 1], bins=hist_nbins, orientation='horizontal',
+    hist_X1.hist(X[features[1]], bins=hist_nbins, orientation='horizontal',
                  color='grey', ec='grey')
     hist_X1.axis('off')
 
     # Histogram for axis X0 (feature 0)
     hist_X0.set_xlim(ax.get_xlim())
-    hist_X0.hist(X[:, 0], bins=hist_nbins, orientation='vertical',
+    hist_X0.hist(X[features[0]], bins=hist_nbins, orientation='vertical',
                  color='grey', ec='grey')
     hist_X0.axis('off')
 
@@ -210,8 +212,8 @@ def make_plot(item_idx):
 
     # zoom-in
     zoom_in_percentile_range = (0, 99)
-    cutoffs_X0 = np.percentile(X[:, 0], zoom_in_percentile_range)
-    cutoffs_X1 = np.percentile(X[:, 1], zoom_in_percentile_range)
+    cutoffs_X0 = np.percentile(X[features[0]], zoom_in_percentile_range)
+    cutoffs_X1 = np.percentile(X[features[1]], zoom_in_percentile_range)
 
     non_outliers_mask = (
         np.all(X > [cutoffs_X0[0], cutoffs_X1[0]], axis=1) &

```

Reply:
Agreed! I will adjust it inside the plotting function as suggested

Reply:
I tried to implement it the way you suggested but it cannot really work like this because of the X in the `make_plot()` function. For all tuples but the first one within the list `distributions`, X will be of type `np.ndarray` (since after scaling you have a `np.ndarray`). Thus slicing has to be used in the plotting function afterwards. One could transform the scaled X back into a pd.DataFrame but I think this is not worth it.

```python
def make_plot(item_idx):
    title, X = distributions[item_idx]
    ax_zoom_out, ax_zoom_in, ax_colorbar = create_axes(title)
    axarr = (ax_zoom_out, ax_zoom_in)
    plot_distribution(axarr[0], X, y, hist_nbins=200,
                      x0_label=feature_mapping[features[0]],
                      x1_label=feature_mapping[features[1]],
                      title="Full data")
```

Reply:
OK so in this case, it is better to not use `as_frame=True` and instead use the attribute `feature_names` stored in the bunch

Reply:
Alright, so I in the new commit I load tha dataset not with `as_frame=True` anymore. The indices for slicing the data are now obtained from the `feature_names` attribute.